Safewhere.Samples.STS/Safewhere.Samples.STS.GenericCredentialsValidator/UsernamePasswordGenericCredentialsValidator.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Web.Mvc;
+using Safewhere.External;
+using Safewhere.External.Authentication;
+
+namespace Safewhere.Samples.STS.GenericCredentialsValidator
+{
+    public class UsernamePasswordGenericCredentialsValidator : IGenericCredentialsValidator
+    {
+        const string ServiceIdentifier = "serviceIdentifier";
+        const string UserName = "username";
+        const string Password = "password";
+        const string AdditionalClaims = "additionalclaim";
+        /// <summary>
+        /// Validate credentials by generic provider
+        /// </summary>
+        /// <param name="cc">Notice that on STs the controller context is irrelevant</param>
+        /// <param name="inputs"></param>
+        /// <param name="logWriter"></param>
+        /// <returns></returns>
+        public CredentialsValidationResult Validate(ControllerContext cc, IDictionary<string, string> inputs, IIdentifyLogWriter logWriter)
+        {
+            var validationResult = ValidateInput(inputs);
+
+            if (validationResult.ResultCode != CredentialsValidationResultCode.Success)
+            {
+                logWriter.WriteError("Validate the input failed. Error code: " + validationResult.ResultCode +
+                                       ". ExternalErrorMessages: " + string.Join("\n-", validationResult.ExternalErrorMessages));
+                validationResult.ShowErrorViewWhenResultCodeIsNotSuccess = false;
+                return validationResult;
+            }
+            var username = inputs[UserName];
+            var serviceIdentifier = inputs[ServiceIdentifier];
+            var logResult = new StringBuilder();
+            logResult.AppendLine(
+                $"Successfully validate generic credentials for username = '{username}' with service identifier ='{serviceIdentifier}'");
+            logResult.AppendLine("");
+
+            foreach (var input in inputs)
+            {
+                if (input.Key.StartsWith(AdditionalClaims, StringComparison.InvariantCultureIgnoreCase))
+                {
+                    logResult.AppendLine($"Additional claims received: type = '{input.Key}' - value ='{input.Value}'");
+                }
+            }
+            logWriter.WriteInformation(logResult); 
+
+            return new CredentialsValidationResult {ResultCode = CredentialsValidationResultCode.Success};
```

Comment:
I think you should demonstrate a failed case. For example, if username is "demowrongusername" and password is "demowrongpassword", this method will return a fail result.